library/core/src/main/java/com/google/android/exoplayer2/text/dvb/DvbParser.java

```diff
@@ -0,0 +1,1569 @@
+            region_version_number                   4     Indicates the version of this region
+            region_fill_flag                        1     If set the region is to be filled region_n-bit_pixel_code clut index
+            reserved                                3
+            region_width                           16     Specifies the horizontal length of this region
+            region_height                          16     Specifies the vertical length of the region
+            region_level_of_compatibility           3     Code that indicates the minimum bithdepth of CLUT
+            region_depth                            3     Identifies the intended pixel depth for this region
+            reserved                                2
+            CLUT_id                                 8     Identifies the family of CLUTs that applies to this region
+            region_8-bit_pixel_code                 8     Specifies the entry of the applied 8-bit CLUT as background colour
+            region_4-bit_pixel-code                 4     Specifies the entry of the applied 4-bit CLUT as background colour
+            region_2-bit_pixel-code                 2     Specifies the entry of the applied 2-bit CLUT as background colour
+            reserved                                2
+            while (processed_length < segment_length) {   list of region objects
+                object_id                          16     Identifies an object that is shown in the region
+                object_type                         2     Identifies the type of object
+                object_provider_flag                2     How this object is provided
+                object_horizontal_position         12     Specifies the horizontal position of the top left pixel of this object
+                reserved                            4
+                object_vertical_position           12     Specifies the vertical position of the top left pixel of this object
+                if (object_type ==0x01 or object_type == 0x02){ UNSUPPORTED
+                    foreground_pixel_code           8
+                    background_pixel_code           8
+                }
+            }
+        }
+        */
+
+    RegionComposition region = new RegionComposition();
+
+    region.pageId = tsStream.readBits(16);
+    int remainingSegmentLength = tsStream.readBits(16);
+    region.regionId = tsStream.readBits(8);
+    region.regionVersionNumber = tsStream.readBits(4);
+    if (tsStream.readBits(1) == 1) {
+      region.flags |= REGION_FILL_FLAG;
+    }
+    tsStream.skipBits(3);
+    region.regionWidth = tsStream.readBits(16);
+    region.regionHeight = tsStream.readBits(16);
+    region.regionLevelOfCompatibility = tsStream.readBits(3);
+    region.regionDepth = tsStream.readBits(3);
+    tsStream.skipBits(2);
+    region.clutId = tsStream.readBits(8);
+    tsStream.skipBits(16);
```

Comment:
Is there a reason region8BitPixelCode/region4BitPixelCode/region2BitPixelCode aren't read and set on the region here? They're currently not set anywhere in code, but they are read (and will always have a value of 0).